integration_tests/test_eip1559.py

```diff
@@ -1,27 +1,70 @@
+from pathlib import Path
+
+import pytest
+from pytest_lazyfixture import lazy_fixture
+
+from .network import setup_custom_cronos
 from .utils import ADDRS, KEYS, send_transaction, w3_wait_for_block
 
+pytestmark = pytest.mark.slow
+
+
+@pytest.fixture(scope="module")
+def custom_cronos(tmp_path_factory):
+    path = tmp_path_factory.mktemp("min-gas-price")
+    yield from setup_custom_cronos(
+        path, 26500, Path(__file__).parent / "configs/min_gas_price.jsonnet"
+    )
+
 
-def adjust_base_fee(parent_fee, gas_limit, gas_used):
+@pytest.fixture(
+    scope="module",
+    params=[
+        lazy_fixture("cronos"),
+        lazy_fixture("geth"),
+        lazy_fixture("cronos_ws"),
+        lazy_fixture("custom_cronos"),
+    ],
+)
+def custom_cluster(request):
+    yield request.param
+
+
+def adjust_base_fee(parent_fee, gas_limit, gas_used, params):
     "spec: https://eips.ethereum.org/EIPS/eip-1559#specification"
-    change_denominator = 8
-    elasticity_multiplier = 2
+    change_denominator = params["base_fee_change_denominator"]
+    elasticity_multiplier = params["elasticity_multiplier"]
     gas_target = gas_limit // elasticity_multiplier
 
     delta = parent_fee * (gas_target - gas_used) // gas_target // change_denominator
-    return parent_fee - delta
+    # https://github.com/crypto-org-chain/ethermint/blob/develop/x/feemarket/keeper/eip1559.go#L104
+    return max(parent_fee - delta, params["min_gas_price"])
+
 
+def get_params(cli):
+    if cli is None:
+        return {
+            "base_fee_change_denominator": 8,
+            "elasticity_multiplier": 2,
+            "min_gas_price": 0,
+        }
+    params = cli.query_params("feemarket")["params"]
+    return {k: int(float(v)) for k, v in params.items()}
 
-def test_dynamic_fee_tx(cluster):
+
+def test_dynamic_fee_tx(custom_cluster, custom_cronos, geth):
```

Comment:
My suggestion is, since geth don't have min-gas-price logic, and this test case do cross-checking with geth, we can leave this test case alone, and do min-gas-price test separately.

Reply:
got it, but we also can't crosscheck value in cronos and geth since these chain [config](https://github.com/ethereum/go-ethereum/blob/master/params/protocol_params.go#L127-L128) can't get changed in geth side.

Reply:
previousely, we keep `change_denominator` and `elasticity_multiplier` the same as geth.

Reply:
the base fee won't the exactly the same, it only checks the same formula applies to both.